packages/phpstan-rules/src/Rules/CheckOptionArgumentCommandRule.php

```diff
@@ -0,0 +1,224 @@
+    private const METHOD_CALL_VALID = [
+        'addoption' => 'getOption',
+        'addargument' => 'getArgument',
+    ];
+
+    /**
+     * @var NodeFinder
+     */
+    private $nodeFinder;
+
+    /**
+     * @var Standard
+     */
+    private $printerStandard;
+
+    public function __construct(NodeFinder $nodeFinder, Standard $printerStandard)
+    {
+        $this->nodeFinder = $nodeFinder;
+        $this->printerStandard = $printerStandard;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param MethodCall $node
+     * @return string[]
+     */
+    public function process(Node $node, Scope $scope): array
+    {
+        if (! $this->isInInstanceOfCommand($node, $scope)) {
+            return [];
+        }
+
+        if (! $this->isInConfigureMethod($node)) {
+            return [];
+        }
+
+        $methodCallIdentifier = $node->name;
+        if (! $methodCallIdentifier instanceof Identifier) {
+            return [];
+        }
+
+        $methodCallName = $methodCallIdentifier->toString();
+        if (! in_array(strtolower($methodCallName), ['addoption', 'addargument'], true)) {
+            return [];
+        }
+
+        return $this->validateInvalidMethodCall($node, $methodCallName);
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(self::ERROR_MESSAGE, [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass extends Command
+{
+    protected function configure(): void
+    {
+        $this->addOption(Option::CATEGORIZE, null, InputOption::VALUE_NONE, 'Group in categories');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $shouldCategorize = (bool) $input->getArgument(Option::CATEGORIZE);
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+class SomeClass extends Command
+{
+    protected function configure(): void
+    {
+        $this->addOption(Option::CATEGORIZE, null, InputOption::VALUE_NONE, 'Group in categories');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $shouldCategorize = (bool) $input->getOption(Option::CATEGORIZE);
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    private function validateInvalidMethodCall(MethodCall $methodCall, string $methodCallName): array
+    {
+        $class = $this->resolveCurrentClass($methodCall);
+        if (! $class instanceof Class_) {
+            return [];
+        }
+
+        /** @var ClassMethod|null $executeClassMethod */
+        $executeClassMethod = $this->getExecuteClassMethod($class);
+        if ($executeClassMethod === null) {
+            return [];
+        }
+
+        $passedArg = $methodCall->args[0]->value;
+        $invalidMethodCall = self::METHOD_CALL_INVALID[strtolower($methodCallName)];
```

Comment:
I'm not sure this rule should handle typos in methods.
Why this section?

Reply:
to compare `getargument` or `getArgumenT`, but I'm ok to use valid method name assumption. I will change.

Reply:
updated to use valid method name assumption.

Reply:
updated constant name to `METHOD_CALL_NOTMATCH` and `METHOD_CALL_MATCH` to avoid ambiguous